vault/resource_consul_role.go

```diff
@@ -0,0 +1,136 @@
+package vault
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func consulRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: consulRoleWrite,
+		Update: consulRoleWrite,
+		Delete: consulRoleDelete,
+		Read:   consulRoleRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the role",
+			},
+			"policy": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The policy document",
+			},
+			"token_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "client",
+				Description: "Consul token type",
+			},
+			"path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     "consul",
+				Description: "Path to the consul secret backend mount. Default is just \"consul\".",
+				ValidateFunc: func(v interface{}, k string) (ws []string, errs []error) {
+					value := v.(string)
+					if strings.HasSuffix(value, "/") {
+						errs = append(errs, fmt.Errorf("path cannot end in '/'"))
+					}
+					return
+				},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old+"/" == new || new+"/" == old
+				},
+			},
+		},
+	}
+}
+
+func consulRoleWrite(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	name := d.Get("name").(string)
+	policy := d.Get("policy").(string)
+	tokenType := d.Get("token_type").(string)
+	userPath := d.Get("path").(string)
+
+	data := map[string]interface{}{
+		"policy":     base64.StdEncoding.EncodeToString([]byte(policy)),
+		"lease":      0,
```

Comment:
This parameter is unnecessary, and should be taken out so that any default value is honored (unless you explicitly want it to be zero always).

Reply:
Also is there a reason not to allow configuration of it by the user?

Reply:
Added a configurable lease parameter.